Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java

```diff
@@ -404,5 +423,61 @@
         httpHeaders.setContentType(MediaType.valueOf(accept));
         return new ResponseEntity<String>("Success", httpHeaders, HttpStatus.OK);
     }
+    
+    
+    /**
+     * Rest interface to get dictionaries related to project
+     * http://<<URL>:<PORT>>/quadriga/auth/rest/<projectid>/dictionaries.json
+     * http://localhost:8080/quadriga/auth/rest/PROJ_bb7ad41b-3e85-4309-b2ff-47d644307b9b/dictionaries.json
+     * 
+     * 
+     * @author Ajay Modi & Bharath Srikantan
+     * @param projectid
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
+
+    @RequestMapping(value = "auth/rest/{projectid}/dictionaries.json", method = RequestMethod.GET, produces = "application/json")
+    public ResponseEntity<String> listProjectDictionaryJson(
+            HttpServletRequest req,
+            @PathVariable("projectid") String projectid, Model model,
+            Principal principal) throws QuadrigaStorageException,
+            QuadrigaException {
+        String userId = principal.getName();
+        List<IProjectDictionary> dictionaryList = null;
+        try {
+            // TODO: listProjectDictionary() is to be changed according to
+            // mapper
+            dictionaryList = projectDictionaryManager.listProjectDictionary(
+                    projectid, userId);
+            
+            dictionaryList.removeAll(Collections.singleton(null));
+            JSONArray ja = new JSONArray();
+            
+            if(dictionaryList!=null){
+                for (IProjectDictionary dictionary : dictionaryList) {
+                    JSONObject j = new JSONObject();
+                    try {
+                        j.put("id", dictionary.getDictionary().getDictionaryId());
+                        j.put("name", dictionary.getDictionary().getDictionaryName());
+                        ja.put(j);
+                    } catch (JSONException e) {
+                        // TODO Auto-generated catch block
+                        throw new QuadrigaException(e.getMessage(), e);
+                    }
+                }
+            }
+                 
+            return new ResponseEntity<String>(ja.toString(), HttpStatus.OK);
+
+        } catch (QuadrigaStorageException e) {
+            throw new QuadrigaStorageException();
+        }

```

Comment:
same as before, try/catch needs to be where it's actually thrown. and proper error handling is needed.